### PR TITLE
Programming expressions edit page examples

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Add the ability to add/edit examples to code docs. Follows [this mock](https://www.figma.com/proto/oJhM1JLs2vMaKvzcrFjZds/Block-Docs?page-id=0%3A1&node-id=102%3A2&viewport=241%2C48%2C0.36&scaling=min-zoom), though adding an image will be added in a future PR.

`OrderableList` is refactored slightly so it can be reused for `ExampleEditor`, along with `ParameterEditor`

Screen Shot 2021-11-08 at 4 08 00 PM

Adding:

https://user-images.githubusercontent.com/46464143/140839940-9aa60daa-f5e1-413c-b02f-b6c873ff3f8b.mov

Rearranging:

https://user-images.githubusercontent.com/46464143/140839971-7bb7822b-31de-4065-873f-e61aeb807e00.mov

Deleting:

https://user-images.githubusercontent.com/46464143/140840021-af581a66-d7b5-4d22-97fd-7aa6f802ee96.mov

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/programming_expressions_controller.rb [hunk 1]
```diff
@@ -70,7 +70,8 @@ def programming_expression_params
       :syntax,
       :return_value,
       :tips,
-      parameters: [:name, :type, :required, :description]
+      parameters: [:name, :type, :required, :description],
+      examples: [:name, :description, :code, :app, :appDisplayType, :appEmbedHeight]
```
Comment (davidsbailey) [inline dashboard/app/controllers/programming_expressions_controller.rb:74]: thanks @tim-dot-org and @bethanyaconnor for getting us onto this pattern!

File: apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import color from '@cdo/apps/util/color';
+import HelpTip from '@cdo/apps/lib/ui/HelpTip';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+
+const APP_DISPLAY_OPTIONS = {
+  directly: 'Embed app with code directly',
+  withCode: 'Display app with code from code field above'
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx:9]: do you have existing examples of these two display styles on https://studio.code.org/docs that you can link to? or is this new functionality we are adding as part of the move?

File: apps/test/unit/lib/levelbuilder/code-docs-editor/ExampleEditorTest.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    expect(appEmbedHeightField.props().value).to.equal('100');
+  });
+
+  it('calls update when app embed height is updated', () => {
+    const wrapper = shallow(<ExampleEditor {...defaultProps} />);
+    const appEmbedHeightField = wrapper.find('input').at(2);
+    appEmbedHeightField.simulate('change', {target: {value: '250'}});
+    expect(updateSpy).to.be.calledOnce.and.calledWith('appEmbedHeight', '250');
+  });
+});
```
Comment (davidsbailey) [inline apps/test/unit/lib/levelbuilder/code-docs-editor/ExampleEditorTest.jsx:97]: nice test coverage 👍

**Review by davidsbailey — APPROVED** [review]
looks good to me!

File: apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx (hunk 2 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx:9]: It is an existing setting on curriculum builder. https://www.codecurricula.com/docs/spritelab/gamelab_jumpTo/ uses "embed code directly". https://www.codecurricula.com/docs/gamelab/camera.y/ uses "Display app with code from code field above"

**Pull request merged**